### ISSUE TITLE
imports

### ISSUE THREAD

**Issue by iframeskills** [marceloverdijk/lesscss-maven-plugin#4]
(thanks for the great plugin! small issue below:)

<pre>
@media screen {
  @import "homepage.less";
  @import "searchresults.less";
}
</pre>

-> fails

<pre>
@media screen {
@import "homepage.less";
@import "searchresults.less";
}
</pre>

-> does not fail.

Whitespace before imports seem to break compilation

<pre>
/*@import "homepage.less";*/
</pre>

also breaks. blocklevel comments and imports don't get along either.

**Comment by marceloverdijk** [comment]
Hi @iframeskills,

I know imports can be a little bit cranky...
The lesscss-maven-plugin uses my lesscss-java library (https://github.com/marceloverdijk/lesscss-java).

Resolving imports is done using a regexp pattern in https://github.com/marceloverdijk/lesscss-java/blob/master/src/main/java/org/lesscss/LessSource.java:

`private static final Pattern IMPORT_PATTERN = Pattern.compile("^(?!\\s*//\\s*)@import\\s+(url\\()?\\s*\"(.+)\\s*\"(\\))?\\s*;.*$", MULTILINE);`

I think this needs a little bit of tweaking...

**Comment by jandebleser** [comment]
Perhaps this is a better regex?

<pre>
^\s*(?!//)\s*@import\s+(url\()?\s*\"(.+)\s*\"(\))?\s*;.*$
</pre>

And as for the block comments, it's always better to filter them out before parsing the file.

<pre>
/\*.*?\*/
</pre>

**Comment by marceloverdijk** [comment]
Good point about filtering block comments before parsing; I will definitely add that!

Honestly, I'm very bad at reg exp, can you maybe explain your changes?

**Comment by jandebleser** [comment]
I just changed the first part 
this:

<pre>
(?!\\s*//\\s*)
</pre>

into this (now with double backslashes):

<pre>
\\s*(?!//)\\s*
</pre>

What it does: The \s\* allows whitespace, and the (?!//) disallows line comments.
What you had before did not allow whitespace at all.

Thanks for the great plugin by the way :)

**Comment by jandebleser** [comment]
And you could even omit the line comment part, since the whitespace rule will already filter it out.
This would then be the regex (with double backslashes):

<pre>
^\\s*@import\\s+(url\\()?\\s*\"(.+)\\s*\"(\\))?\\s*;.*$
</pre>

**Comment by candrews** [comment]
I'm seeing this issue as well - do you have a guess when this fix may be made? Thanks!

**Comment by alanshaw** [comment]
+1

I just got caught out as well.

**Comment by marceloverdijk** [comment]
I was waiting for a next less version...

**Comment by disolovyov** [comment]
You might as well incorporate single quotes into the regex, as this is also valid:

``` css
@import 'file.less';
```

Regexes aside, have you considered not tying lesscss-java releases strictly to less.js?

**Comment by rweng** [comment]
+1 to singe quotes in regex. This broke my build.

**Comment by MattSenter** [comment]
+1 This just got me too. Was about to file an issue after tracking it down, only to realize I should have read this one first! =)

**Comment by cpopov** [comment]
I will look into it. Simple quote imports is fixed for 1.3.3. I need to reproduce the issue with white spaces.

**Comment by cpopov** [comment]
normally lesscss.js manages imports. But I am not sure how it will work in a Rhino environment. I am just thinking that if lesscss.js manages imports, maybe we shouldn't do a pre-processing with regexps.
